tests/k8s/Vagrantfile

```diff
@@ -79,6 +87,28 @@ Vagrant.configure(2) do |config|
         s.inline = "sudo sed -i '/tty/!s/mesg n/tty -s \\&\\& mesg n/' /root/.profile"
     end
 
+    # Set up k8s0 as Docker registry, running newer version of Docker to support staged
+    # builds and not running k8s.  This VM will be idle after k8s[12] have pulled the
+    # newly built cilium image.
+    config.vm.define "k8s0#{$build_id_name}" do |s|
+        s.vm.hostname = "k8s-0"
```

Comment:
Why not give this a more descriptive name? It's transient, as you say, and it doesn't run k8s. envoy-builder-registry (or something) would make it clearer if, for example, we leak VMs on a build machine one day.

Reply:
Most likely I initially thought I could use the same loop that loops from 1 to 2 to also configure the new VM. That turned out not to be the case, so I'll experiment with a more descriptive name.

Either way, the certs need to be regenerated as the IP address of the registry node needs to be in a cert. I guess we could use a separate, completely new cert for the registry node and leave the k8s cluster cert untouched, but that would have been more work and more new files in the repo. I'm leaning for someone more experienced working with certs fixing this up later, though.

Reply:
Ah, I didn't realise there was this much interrrelation between the certs and nodes. I see you renamed it below which more than addresses my primary concern. Thanks.